lib/blacklight/component.rb

```diff
@@ -8,6 +8,10 @@ def compiler
         @__vc_compiler ||= EngineCompiler.new(self)
       end
       # rubocop:enable Naming/MemoizedInstanceVariableName
+
+      def with_individual(object, **kwargs)
```

Comment:
I'm not sure I understand why we want or need this method.  This makes the logic more challenging to follow.  Why not just use the collection parameter outrightright?

Reply:
I'm basically trying to accommodate a configured document component called e.g. `DocComponent` with idiomatic parameters (`doc:, ...`) being usable here without overriding the calling partial.

Reply:
If we require `presenter:` and never take `document:` we will be working against ViewComponent's iteration (at least without specifiying the collection parameter, and then we're back to square one on subclasses). I think we want to strive to be as "normal" as possible about the ViewComponent stuff, and be able to rely on their documentation. But I'm also open to ideas about how to do this!